metrics-operator/api/v1alpha1/keptnmetric_types.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KeptnMetricSpec defines the desired state of KeptnMetric
+type KeptnMetricSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of KeptnMetric. Edit keptnmetric_types.go to remove/update
+	Foo string `json:"foo,omitempty"`
+}
+
+// KeptnMetricStatus defines the observed state of KeptnMetric
+type KeptnMetricStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// KeptnMetric is the Schema for the keptnmetrics API
+type KeptnMetric struct {
```

Comment:
the v1alpha1 version had to be added here as well, since the CRD containing the v1alpha2 version could not be applied to a cluster where the previous version is already there